File: server/src/main/java/ru/mityugov/budget_planner/repository/entity/IncomeEntity.java
package ru.mityugov.budget_planner.repository.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDate;

@Setter
@Getter
@Entity
@Table(name = "income")
public class IncomeEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Long id;

    @Column(name = "title")
    private String title;

    @Column(name = "date")
    private LocalDate localDate;

    @Column(name = "amount")
    private int amount;
}

File: src/main/kotlin/ru/mityugov/budgetplanner/infrastructure/entity/IncomeEntity.kt
package ru.mityugov.budgetplanner.infrastructure.entity

import jakarta.persistence.*
import java.time.LocalDate
import java.util.*

@Entity
@Table(name = "income")
class IncomeEntity {

    @Id @GeneratedValue(strategy = GenerationType.IDENTITY) var id: Long? = null
    val name: String
    val uuid: UUID
    val amount: Float
    val date: LocalDate
    val isDone: Boolean
    @ManyToMany
    @JoinTable(
        name = "category_income_link",
        joinColumns = [JoinColumn(name = "income_id")],
        inverseJoinColumns = [JoinColumn(name = "category_id")]
    )
    val categories: MutableSet<IncomeCategoryEntity>
    @ManyToMany
    @JoinTable(
        name = "user_income_link",
        joinColumns = [JoinColumn(name = "income_id")],
        inverseJoinColumns = [JoinColumn(name = "user_id")]
    )
    val owners: MutableSet<UserEntity>

    constructor(
        uuid: UUID,
        name: String,
        amount: Float,
        date: LocalDate,
        isDone: Boolean,
        categories: MutableSet<IncomeCategoryEntity>,
        owners: MutableSet<UserEntity>
    ) {
        this.name = name
        this.amount = amount
        this.date = date
        this.isDone = isDone
        this.categories = categories
        this.owners = owners
        this.uuid = uuid
    }

}